src/file-selector.ts

```diff
@@ -111,8 +111,8 @@ function toFilePromises(item: DataTransferItem) {
   // Safari supports dropping an image node from a different window and can be retrieved using
   // the DataTransferItem.getAsFile() API
   // NOTE: FileSystemEntry.file() throws if trying to get the file
-  if (entry && entry.isDirectory) {
-    return fromDirEntry(entry) as any;
+  if (entry && isFileSystemDirectoryEntry(entry)) {
```

Comment:
I mean, same as before, a function call is not really necessary here.

Reply:
It's more about type soundness, see https://github.com/react-dropzone/file-selector/pull/112#discussion_r1840708105